core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -106,6 +105,7 @@ public void inject(AdvancedCache<Object, Object> cache, @ComponentName(CACHE_MAR
       this.persistenceExecutor = persistenceExecutor;
       this.byteBufferFactory = byteBufferFactory;
       this.marshalledEntryFactory = marshalledEntryFactory;
+      this.cacheStoreFactoryRegistry = cache.getComponentRegistry().getGlobalComponentRegistry().getComponent(CacheStoreFactoryRegistry.class);
```

Comment:
Use the @DefaultFactoryFor annotation on a class in the org.infinispan.factories package